src/processing/script/header.ts

```diff
@@ -33,20 +34,23 @@ const HEADER: string = `
                 ${INSTRUCTION.getProxyUrl} = window.${INSTRUCTION.getProxyUrl},
                 ${INSTRUCTION.restArray} = window.${INSTRUCTION.restArray},
                 ${INSTRUCTION.restObject} = window.${INSTRUCTION.restObject};
-        }
     } else {
-        var ${INSTRUCTION.getLocation} = function(l){return l},
-            ${INSTRUCTION.setLocation} = function(l,v){return l = v},
-            ${INSTRUCTION.setProperty} = function(o,p,v){return o[p] = v},
-            ${INSTRUCTION.getProperty} = function(o,p){return o[p]},
-            ${INSTRUCTION.callMethod} = function(o,p,a){return o[p].apply(o,a)},
-            ${INSTRUCTION.getEval} = function(e){return e},
-            ${INSTRUCTION.processScript} = function(s){return s},
-            ${INSTRUCTION.processHtml} = function(h){return h},
-            ${INSTRUCTION.getPostMessage} = function(w,p){return arguments.length===1?w.postMessage:p},
-            ${INSTRUCTION.getProxyUrl} = function(u,d){return u},
-            ${INSTRUCTION.restArray} = function(a,i){return Array.prototype.slice.call(a, i)},
-            ${INSTRUCTION.restObject} = function(o,p){var k=Object.keys(o),n={};for(var i=0;i<k.length;++i)if(p.indexOf(k[i])<0)n[k[i]]=o[k[i]];return n};
+        if (typeof ${INSTRUCTION.getProperty} === 'undefined')
+            var ${INSTRUCTION.getLocation} = function(l){return l},
+                ${INSTRUCTION.setLocation} = function(l,v){return l = v},
+                ${INSTRUCTION.setProperty} = function(o,p,v){return o[p] = v},
+                ${INSTRUCTION.getProperty} = function(o,p){return o[p]},
+                ${INSTRUCTION.callMethod} = function(o,p,a){return o[p].apply(o,a)},
+                ${INSTRUCTION.getEval} = function(e){return e},
+                ${INSTRUCTION.processScript} = function(s){return s},
+                ${INSTRUCTION.processHtml} = function(h){return h},
+                ${INSTRUCTION.getPostMessage} = function(w,p){return arguments.length===1?w.postMessage:p},
+                ${INSTRUCTION.getProxyUrl} = function(u,d){return u},
+                ${INSTRUCTION.restArray} = function(a,i){return Array.prototype.slice.call(a, i)},
+                ${INSTRUCTION.restObject} = function(o,p){var k=Object.keys(o),n={};for(var i=0;i<k.length;++i)if(p.indexOf(k[i])<0)n[k[i]]=o[k[i]];return n};
+        
+        if (typeof importScripts !== "undefined" && /\\[native code]/g.test(importScripts.toString()))
```

Comment:
Can we simplify template and put this code to the `worker-hammerhead.js`?

Reply:
https://github.com/DevExpress/testcafe-hammerhead/issues/2404